classes/Entities/CourseCategory.php

```diff
@@ -2,7 +2,28 @@
 
 namespace Avado\MoodleAbstractionLibrary\Entities;
 
+use Avado\MoodleAbstractionLibrary\Entities\BaseModel;
+use local\helpdesk\Entities\HelpdeskProduct;
+use local\helpdesk\Entities\HelpdeskProductCourses;
+
 class CourseCategory extends BaseModel
 {
     protected $table = 'course_categories';
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function subCategory(){
+        return $this->hasMany(CourseCategory::class,'parent','id');
+    }
+
+    /**
+     * Get the HelpdeskProductCourses that the Coursecategory belong to
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function helpdeskProductCourses(){
```

Comment:
Not a required change but just so you are aware, you dont have to name these functions after the model you are linking them to. If a there is a more readable descriptive name that works better you can go with that.

Reply:
Thanks will know for the future.
I will still try to find a better name for it